src/main/kotlin/org/rust/ide/hints/RsExpressionTypeProvider.kt

```diff
@@ -20,18 +26,43 @@ class RsExpressionTypeProvider : ExpressionTypeProvider<PsiElement>() {
     override fun getErrorHint(): String = "Select an expression!"
 
     override fun getExpressionsAt(pivot: PsiElement): List<PsiElement> =
-        pivot.ancestors
+        pivot.findExpansionElementOrSelf().contexts
             .takeWhile { it !is RsItemElement }
             .filter { it is RsExpr || it is RsPat || it is RsPatField }
+            .mapNotNull { it.wrapIfInMacroExpansion() }
             .toList()
 
     override fun getInformationHint(element: PsiElement): String {
-        val type = when (element) {
-            is RsExpr -> element.type
-            is RsPat -> element.type
-            is RsPatField -> element.type
-            else -> error("Unexpected element type: $element")
-        }
+        val type = getType(element)
         return type.toString().escaped
     }
+
+    private fun getType(element: PsiElement): Ty = when (element) {
+        is MyFakePsiElement -> getType(element.elementInMacroExpansion)
+        is RsExpr -> element.type
+        is RsPat -> element.type
+        is RsPatField -> element.type
+        else -> error("Unexpected element type: $element")
+    }
+}
+
+private fun PsiElement.wrapIfInMacroExpansion(): PsiElement? {
```

Comment:
Could we come up with better name?
For example, `wrapExpandedElements`?
Also, I'd prefer to add a short comment why we need it at all